rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/rascal/RascalTextDocumentService.java

```diff
@@ -382,7 +385,44 @@ public CompletableFuture<WorkspaceEdit> rename(RenameParams params) {
                 ITree cursorTree = findQualifiedNameUnderCursor(focus);
                 return rascalServices.getRename(TreeAdapter.getLocation(cursorTree), focus, workspaceFolders, facts::getPathConfig, params.getNewName()).get();
             })
-            .thenApply(t -> DocumentChanges.translateDocumentChanges(this, t));
+            .thenApply(t -> {
+                showMessages((ISet) t.get(1));
+                return DocumentChanges.translateDocumentChanges(this, (IList) t.get(0));
+            });
+    }
+
+    private <T extends ICollection<T>> void showMessages(ICollection<T> messages) {
+        for (var msg : messages) {
+            client.showMessage(mapMessage((IConstructor) msg));
+        }
+    }
+
+    private MessageParams mapMessage(IConstructor message) {
```

Comment:
can we make a better name than `mapMessage`? (and similar for `showMessage`)

Reply:
Renamed to `setMessageParams`. `showMessage` comes from the LSP4J API, so we cannot rename that. Or do you mean `showMessages`?